eclipse-theia/theia
# [electron]: Moved the menu creation/update to electron-main.

#### What it does


This PR moves the electron menu creation/update to the electron-main process.

OK but why?
 - Electron `remote` has been deprecated and will be moved outside of electron, (https://github.com/electron/electron/issues/21408)
 - `getCurrentWindow` is a synchronous call and blocks the thread. This is problematic when Theia (or its customization) opens with several browser windows at startup,
 - This will help to remove the rest of the Electron `remote` code from Theia and use IPC between the main and the renderers.

One of my top priorities during the development was to reduce the amount of breaking changes. Please chime in and propose changes. Feedback is welcome. Thank you! 🧁

#### How to test
 - Start the Electron app in debug mode, and check the console how the context menu items are cleaned up on the main process,
 - TBD,

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
